src/test/java/org/folio/dao/DataImportQueueDaoTest.java

```diff
@@ -37,35 +41,54 @@ public class DataImportQueueDaoTest  {
 
   @InjectMocks
   DataImportQueueItemDao queueItemDaoImpl = new DataImportQueueItemDaoImpl();
-  
 
   UUID storedItemUUID;
+
   @Before
   public void setUp() {
-    MockitoAnnotations.initMocks(this);
+    MockitoAnnotations.openMocks(this);
     when(postgresClientFactory.getInstance()).thenReturn(postgresClient);
   }
-  @Test 
+
+  @Test
   public void shouldAddQueueItem(TestContext context) {
     // given
-    when(postgresClient.execute(anyString(), any(Tuple.class)))
-      .thenReturn(Future.succeededFuture(new LocalRowSet(1)));
+    doAnswer((InvocationOnMock invocation) -> {
+        Promise<RowSet<Row>> promise = invocation.getArgument(2);
+        promise.complete(new LocalRowSet(1));
+        return null;
+      })
+      .when(postgresClient)
+      .execute(
+        anyString(),
+        any(Tuple.class),
+        ArgumentMatchers.<Promise<RowSet<Row>>>any()
```

Comment:
It looks really ugly, but this is the way to make an `any()` that accepts a generic (since we can't do `.class` on a generic, otherwise we'd be able to just use the syntax like `Tuple.class` uses.